src/ResourceManager/Network/Commands.Network/ApplicationGateway/CustomError/GetAzureApplicationGatewayCustomerErrorCommand.cs

```diff
@@ -0,0 +1,78 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Network.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Network
+{
+    [Cmdlet("Get", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "ApplicationGatewayCustomError"),
+        OutputType(typeof(PSApplicationGatewayCustomError))]
+    public class GetAzureApplicationGatewayCustomerErrorCommand : NetworkBaseCmdlet
+    {
+        [Parameter(
+                Mandatory = false,
+                HelpMessage = "Status code of the application gateway customer error.")]
+        [ValidateNotNullOrEmpty]
+        public string StatusCode { get; set; }
+
+        [Parameter(
+             Mandatory = false,
+             ValueFromPipeline = true,
+             HelpMessage = "The Application Gateway")]
+        public PSApplicationGateway ApplicationGateway { get; set; }
```

Comment:
The user should have to specify one or the other and not both or none of them.. right?

Correct way to handle that is to put the params `applicationgateway` and `httplistener` in their own parameter sets and mark them as mandatory.